core/src/main/scala/neotypes/implicits/mappers/ResultMappers.scala

```diff
@@ -147,14 +147,24 @@ trait ResultMappers extends ValueMappers {
               }
 
             val decodedHead = head.to(fieldName, convertedValue.find(_._1 == fieldName).map(_._2))
-            decodedHead.flatMap(v => tail.to(convertedValue, typeHint).map(t => labelled.field[K](v) :: t))
+
+            tail match{
+              case HNilResultMapper if errors.nonEmpty =>
+                Left(MultipleIncoercibleException(errors.toList.reverse))
+              case _ =>
+                decodedHead match{
+                  case Left(th) =>
+                    tail.to(convertedValue, typeHint, th +: errors).map(t => labelled.field[K](null.asInstanceOf[H]) :: t)
+                  case Right(v) => tail.to(convertedValue, typeHint, errors).map(t => labelled.field[K](v) :: t)
+                }
```

Comment:
I would format it like:

```scala
decodedHead match {
  case Left(th) =>
     tail.to(convertedValue, typeHint, th :: errors).map(t => labelled.field[K](null.asInstanceOf[H]) :: t)

  case Right(v) =>
    tail.to(convertedValue, typeHint, errors).map(t => labelled.field[K](v) :: t)
}
```

Also, both `null` and `asInstanceOf` are code smells.
There should be a better way to make it typecheck.

---

_(I will try to play with the code on an editor to understand why you did that, but that will probably take me a couple of days)_